pkg/kubectl/resize.go

```diff
@@ -70,6 +83,19 @@ type ReplicationControllerResizer struct {
 	client.Interface
 }
 
+func CastResizeAsCondition(r Resizer, precondition *ResizePrecondition, namespace, name string, count uint) wait.ConditionFunc {
+	return func() (bool, error) {
+		switch _, err := r.Resize(namespace, name, precondition, count); err.(type) {
+		case nil:
+			return true, nil
+		case ControllerUpdateError:
```

Comment:
Ah, I see. I'm pretty neutral on whether that is important (should be rare? we hope?), but if you want to do that, perhaps it's better to make a ControllerResizeError type with an enum telling you if the error happened during Get or Update?